delugeclient.go

```diff
@@ -292,6 +296,9 @@ func (c *Client) Close() error {
 const Deluge2ProtocolVersion = 1
 
 func (c *Client) rpc(methodName string, args rencode.List, kwargs rencode.Dictionary) (*DelugeResponse, error) {
+	c.mu.Lock()
```

Comment:
Why is this added here?

Reply:
I read from your description:
> Use mutual exclusion lock to fix mismatch c.serial in goroutines.

But this seems unrelated to adding the two methods and a pretty big change. What is the reason you added this? What did you discover about the Deluge Daemon RPC?

Reply:
Deluge Daemon is OK. But I found a lot of error from client log likes:
```
request/response serial id mismatch: got 886067 but 886070 expected
```
I have 2-3 goroutines to read response from one Client instance at the same time, but the time spent of responses are not consistent, I think here is why mismatch serial number.
No better solution for now.
btw more tests will up later.

Reply:
Yes, it's another question.